modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/AbstractGeoBucketAggregationIntegTest.java

```diff
@@ -88,14 +90,13 @@ protected void prepareGeoShapeIndexForAggregations(final Random random) throws E
             final Geometry geometry = RandomGeoGeometryGenerator.randomGeometry(random);
             final GeoShapeDocValue geometryDocValue = GeoShapeDocValue.createGeometryDocValue(geometry);
             // make sure that there is 1 shape is intersecting with the bounding box
-            if (!isShapeIntersectingBB) {
-                isShapeIntersectingBB = geometryDocValue.isIntersectingRectangle(boundingRectangleForGeoShapesAgg);
-                if (!isShapeIntersectingBB && i == NUM_DOCS - 1) {
-                    continue;
-                }
+            isShapeIntersectingBB = geometryDocValue.isIntersectingRectangle(boundingRectangleForGeoShapesAgg);
```

Comment:
I think the previous code is correct here. This new code will try to generate intersecting rectangle for the last document even if an intersecting rectangle was added already in previous iteration.

Reply:
Correct. I fixed that.